src/Transport/Addressing/Sanitization/Strategies/ValidateAndHashIfNeeded.cs

```diff
@@ -1,17 +1,116 @@
 ﻿namespace NServiceBus.AzureServiceBus.Addressing
 {
     using System;
+    using System.Text.RegularExpressions;
     using Settings;
 
     public class ValidateAndHashIfNeeded : ISanitizationStrategy
     {
         readonly ReadOnlySettings settings;
 
+        // Entity segments can contain only letters, numbers, periods (.), hyphens (-), and underscores (-), paths can contain slashes (/)
+        Regex queueAndTopicNameRegex = new Regex(@"^[^\/][0-9A-Za-z_\.\-\/]+[^\/]$");
+
+        // Except for subscriptions and rules, these cannot contain slashes (/)
+        Regex subscriptionAndRuleNameRegex = new Regex(@"^[0-9A-Za-z_\.\-]+$");
+
+        Regex sanitizationRegex = new Regex(@"[^a-zA-Z0-9\-\._]");
```

Comment:
Documentation on [`ValidateAndHashIfNeeded`](http://docs.particular.net/nservicebus/azure-service-bus/sanitization#automated-sanitization-versions-7-and-above) will need to be updated @yvesgoeleven